tf-modules/ingest/iam.tf

```diff
@@ -0,0 +1,98 @@
+data "aws_iam_policy_document" "application_autoscaling_assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["application-autoscaling.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "states_assume_role_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    principals {
+      type        = "Service"
+      identifiers = ["states.${data.aws_region.current.name}.amazonaws.com"]
+    }
+  }
+}
+
+# Scaling role
+
+resource "aws_iam_role" "scaling" {
+  name                 = "${var.prefix}-scaling-role"
+  assume_role_policy   = data.aws_iam_policy_document.application_autoscaling_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+  # TODO Re-enable once IAM permissions have been fixed
+  # tags                 = local.default_tags
+}
+
+data "aws_iam_policy_document" "scaling_policy" {
+  statement {
+    actions = [
+      "application-autoscaling:*",
+      "cloudwatch:DeleteAlarms",
+      "cloudwatch:DescribeAlarms",
+      "cloudwatch:GetMetricStatistics",
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:SetAlarmState",
+      "dynamodb:DescribeTable",
+      "dynamodb:UpdateTable",
+      "ecs:DescribeServices",
+      "ecs:UpdateService",
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_role_policy" "scaling" {
+  role   = aws_iam_role.scaling.id
+  policy = data.aws_iam_policy_document.scaling_policy.json
+}
+
+# Step role
+
+resource "aws_iam_role" "step" {
+  name                 = "${var.prefix}-steprole"
+  assume_role_policy   = data.aws_iam_policy_document.states_assume_role_policy.json
+  permissions_boundary = var.permissions_boundary_arn
+  # TODO Re-enable once IAM permissions have been fixed
+  # tags                 = local.default_tags
+}
+
+data "aws_iam_policy_document" "step_policy" {
+  statement {
+    actions = [
+      "lambda:InvokeFunction",
+      "ecr:*",
+      "cloudtrail:LookupEvents",
+      "ecs:RunTask",
+      "ecs:StopTask",
+      "ecs:DescribeTasks",
+      "autoscaling:Describe*",
+      "cloudwatch:*",
+      "logs:*",
+      "sns:*",
+      "iam:GetPolicy",
+      "iam:GetPolicyVersion",
+      "iam:GetRole",
+    ]
+    resources = ["*"]
+  }
+
+  statement {
+    actions = [
+      "events:DescribeRule",
+      "events:PutRule",
+      "events:PutTargets"
+    ]
+    # TODO Where is this being created?
+    resources = ["arn:aws:events:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:rule/StepFunctionsGetEventsForECSTaskRule"]
```

Comment:
It looks like it's related to a MAAP PR that came in about 7 months ago - https://github.com/nasa/cumulus/pull/784. @markdboyd brought it in and @laurenfrederick approved it, so I'm going to defer to you guys.

Reply:
Hmm. Looks like that rule ID was just hard-coded in that PR, so that permission only was useful to people who had that rule deployed. 

I'm wondering how/if we support this in a world where there are no "global" IAM roles and Lambdas have minimum permissions. This seems related to Step functions running ECS activities and I generally don't know how we are managing permissions for that in Terraform.

Reply:
Is there any necessary follow up for this PR on that rule @markdboyd? Should we be perpetuating that permission?

Reply:
I'm inclined to say that we cut it for now. It feels odd to have a hard-coded reference to a resource that we don't actually manage. If we need to support this, we should probably do so a different way

Reply:
"Cut" as in "just delete this IAM statement"?

Reply:
Yes